src/astro/files/type/__init__.py

```diff
@@ -44,7 +44,16 @@ def get_filetype(filepath: Union[str, pathlib.PosixPath]) -> FileTypeConstants:
     if isinstance(filepath, pathlib.PosixPath):
         extension = filepath.suffix[1:]
     else:
-        extension = filepath.split(".")[-1]
+        extension = ""
+        tokenized_path = filepath.split(".")
+        if len(tokenized_path) > 1:
+            extension = tokenized_path[-1]
+
+    if extension == "":
+        raise ValueError(
+            f"Missing file extension, cannot determine filetype from path '{filepath}'."
+            f" If this path contains pattern pass 'filetype' param."
```

Comment:
what do you mean by "pattern" here?

Reply:
@kaxil I think this means that we can't autoparse the type so the user needs to explicitly state it. I changed the comment to reflect this.